services/user.py

```diff
@@ -0,0 +1,67 @@
+from django.contrib.auth import get_user_model
+from django.db import IntegrityError, transaction
+
+User = get_user_model()
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None
+) -> None:
+    extra_fields = {
+        "email": email,
+        "first_name": first_name,
+        "last_name": last_name
+    }
+    extra_fields = {
+        key: value for key, value in extra_fields.items() if value is not None
+    }
+
+    try:
+        User.objects.create_user(
+            username=username,
+            password=password,
+            **extra_fields
+        )
+    except IntegrityError:
+        raise ValueError(
+            f"A user with the username '{username}' already exists."
+        )
```

Comment:
The error message raised in the `ValueError` could be more informative by including the email if it's also a unique field, as this would help in debugging cases where the email might already exist.